### PR TITLE
[IMPROVE] Added identification on calls to/from existing contacts

### PR DESCRIPTION
**Pull request by aleksandernsilva**
## Proposed changes (including videos or screenshots)

Before: 
Screen Shot 2022-07-02 at 01 50 52

After:
Screen Shot 2022-07-21 at 16 00 27

### CODE REVIEW

File: apps/meteor/client/components/FilterByText.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const FilterByText = ({
 	}, []);
 
 	return (
-		<Box mb='x16' is='form' onSubmit={handleFormSubmit} display='flex' flexDirection={shouldFiltersStack ? 'column' : 'row'} {...props}>
+		<Box mb='x16' is='form' onSubmit={handleFormSubmit} display='flex' flexDirection={shouldFiltersStack ? 'column' : 'row'}>
```
Comment (tiagoevanp) [inline apps/meteor/client/components/FilterByText.tsx:49]: Have you checked if the spread props is not necessary in every usage case?

**Review by tiagoevanp — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/FilterByText.tsx (hunk 1 above)
Comment (aleksandernsilva) [reply apps/meteor/client/components/FilterByText.tsx:49]: Considering the properties this component accepts, there's nothing relevant to spread.

**Review by tiagoevanp — APPROVED** [review]

**Review by tiagoevanp — APPROVED** [review]

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const createContact = (phone: string, data: Pick<ILivechatVisitor, 'name'> | null): Contact => ({
+	phone,
+	name: data?.name || '',
+});
+
+export const useOmnichannelContact = (phone: string): Contact => {
+	const getContactBy = useEndpoint('GET', '/v1/omnichannel/contact.search');
+	const safePhone = parseOutboundPhoneNumber(phone);
+	const [contact, setContact] = useState<Contact>({ phone: safePhone, name: safePhone });
```
Comment (tassoevan) [inline apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts:20]: Handle this state using `@tanstack/react-query`.

**Review by tassoevan — DISMISSED** [review]

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+			return createContact(phone, data.contact);
+		},
+		[getContactBy],
+	);
+
+	useEffect(() => {
+		getContactByPhone(safePhone).then(setContact);
+	}, [safePhone, getContactByPhone]);
+
+	return contact;
```
Comment (ggazzo) [inline apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts:34]: use `useQuery` here

File: apps/meteor/client/sidebar/sections/actions/OmnichannelCallToggleLoading.tsx [hunk 4]
```diff
@@ -4,5 +4,14 @@ import React, { ReactElement } from 'react';
 
 export const OmnichannelCallToggleLoading = ({ ...props }): ReactElement => {
 	const t = useTranslation();
-	return <Sidebar.TopBar.Action icon='phone' data-title={t('Loading')} disabled {...props} />;
+	return (
+		<Sidebar.TopBar.Action
+			icon='phone'
+			data-title={t('Loading')}
+			data-qa-id='omncVoipToggleButton'
```
Comment (ggazzo) [inline apps/meteor/client/sidebar/sections/actions/OmnichannelCallToggleLoading.tsx:11]: i'd seen a lot of data-qa-id duplicated, makes me questioning this usage

File: apps/meteor/client/sidebar/footer/voip/VoipFooter.tsx [hunk 5]
```diff
@@ -132,6 +135,7 @@ export const VoipFooter = ({
 								small
 								square
 								danger
+								data-qa-id='omncVoipRejectButton'
```
Comment (ggazzo) [inline apps/meteor/client/sidebar/footer/voip/VoipFooter.tsx:138]: even data-qa been more permissive then `id` itself makes no sense to me have multiple data-qa-id with same value

File: apps/meteor/client/sidebar/footer/voip/VoipFooter.tsx [hunk 6]
```diff
@@ -143,7 +147,7 @@ export const VoipFooter = ({
 							</Button>
 						)}
 						{callerState === 'OFFER_RECEIVED' && (
-							<Button title={tooltips.endCall} small square danger onClick={callActions.reject}>
+							<Button title={tooltips.endCall} data-qa-id='omncVoipRejectButton' small square danger onClick={callActions.reject}>
```
Comment (ggazzo) [inline apps/meteor/client/sidebar/footer/voip/VoipFooter.tsx:150]: `data-qa-..` duplicated 🤔

File: apps/meteor/client/components/FilterByText.tsx [hunk 7]
```diff
@@ -46,7 +46,7 @@ const FilterByText = ({
 	}, []);
 
 	return (
-		<Box mb='x16' is='form' onSubmit={handleFormSubmit} display='flex' flexDirection={shouldFiltersStack ? 'column' : 'row'} {...props}>
```
Comment (ggazzo) [inline apps/meteor/client/components/FilterByText.tsx:49]: why did you remove this?

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/FilterByText.tsx (hunk 7 above)
Comment (aleksandernsilva) [reply apps/meteor/client/components/FilterByText.tsx:49]: We were sending non html properties to the element, which was throwing warnings. Looking into this component's interface there's no valid attribute to be spreaded, so I removed it.

**Review by MartinSchoeler — APPROVED** [review]
The refactor to use `@tanstack/react-query` and the changes to `data-qa` are OK!

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	const {
+		data: contact,
+		isLoading,
+		isError,
+	} = useQuery(['getContactsByPhone', phone], async (): Promise<Contact> => {
+		const { contact } = await getContactBy({ phone });
+		return createContact(phone, contact);
+	});
+
+	return isLoading || isError || !contact ? defaultContact : contact;
```
Comment (ggazzo) [inline apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts:32]: I don't like to omit hook states (loading/error). Why not return the entire query? instead of the default value?

File: apps/meteor/tests/e2e/omnichannel-voip-footer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+						{ _id: 'VoIP_Management_Server_Name', value: 'OminiAsterisk' },
+						{ _id: 'VoIP_Management_Server_Username', value: 'sales.rocket.chat' },
+						{ _id: 'VoIP_Management_Server_Password', value: 'rocket@123' },
+						{ _id: 'VoIP_Server_Name', value: 'OmniAsterisk' },
+						{ _id: 'VoIP_Server_Websocket_Path', value: 'wss://omni-asterisk.dev.rocket.chat/ws' },
+					],
+					{
+						twoFactorCode: 'b6769a5ae0a6071ecabbe868dbdfa925f856c2bb3d910f93cb39479c64ca221e',
+						twoFactorMethod: 'password',
+					},
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel-voip-footer.spec.ts:68]: ow my, this is something we should keep secret under env vars

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts (hunk 8 above)
Comment (aleksandernsilva) [reply apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts:32]: @ggazzo The idea is that if you are using this hook you can trust it to always return a contact. You don't have to worry about what's happening behind the scenes (loading/error). I can return those states, no problem, but at least for now I don't see a scenario where I would need them. I believe a good compromise is returning and something like `{ isLoading: boolean; isError: boolean; contact: Contact }`, wdyt?

File: apps/meteor/tests/e2e/omnichannel-voip-footer.spec.ts [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+};
+
+const createPageObjects = (page: Page) => ({
+	page,
+	omncSection: new OmnichannelSection(page),
+	dialpadModal: new OmnichannelDialpadModal(page),
+	voipFooter: new OmnichannelVoipFooter(page),
+});
+
+test.skip(!IS_EE, 'Omnichannel Voip Footer > Enterprise Only');
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel-voip-footer.spec.ts:23]: I could not found the test case you are skipping

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+		isError,
+	} = useQuery(['getContactsByPhone', phone], async (): Promise<Contact> => {
+		const { contact } = await getContactBy({ phone });
+		return createContact(phone, contact);
+	});
+
+	return {
+		contact: contact || defaultContact,
+		isLoading,
+		isError,
```
Comment (ggazzo) [inline apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts:41]: c'mon i asked you to return the query to actually use it

**Review by ggazzo — APPROVED** [review]
I'm going to ask you to pay more attention when creating custom hooks, ok it's custom, but I really hate it when a query becomes a default data and the application no longer knows if it's loading or if the data is undefined.

however this is an alteration that does not harm anyone. But avoid spreading it around.

**Review by ggazzo — CHANGES_REQUESTED** [review]
Can ask you to check the two TODO: that I'd inserted? thanks

File: apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts (hunk 11 above)
Comment (aleksandernsilva) [reply apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts]: > I'm going to ask you to pay more attention when creating custom hooks, ok it's custom, but I really hate it when a query becomes a default data and the application no longer knows if it's loading or if the data is undefined.
> 
> however this is an alteration that does not harm anyone. But avoid spreading it around.

I don't seem to understand what you expect from this return. If you're available i'd love to talk and get to know what's on your mind.

**Review by ggazzo — DISMISSED** [review]

**Review by tiagoevanp — DISMISSED** [review]

**Review by tiagoevanp — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

File: apps/meteor/ee/client/hooks/useVoipClient.ts [hunk 12]
```diff
@@ -34,6 +35,10 @@ export const useVoipClient = (): UseVoipClientResult => {
 
 	const isEE = useHasLicenseModule('voip-enterprise');
 
+	useEffect(() => {
+		setVoipEnabled(settingVoipEnabled);
+	}, [settingVoipEnabled, setVoipEnabled]);
+
```
Comment (ggazzo) [inline apps/meteor/ee/client/hooks/useVoipClient.ts:41]: Can you explain me whatta hell is happening here?

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Review by tiagoevanp — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]
